### PR TITLE
Merging feature/authorize-api to develop

### PR DESCRIPTION
**Pull request by kn9ts**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,7 +1,13 @@
 CLOUDINARY_URL = cloudinary://258585947589249:ZUACNOFbFG3iiQ83XCT1smXdroI@vvida
```
Comment (andela-gukpere) [inline .env]: You should not stage the .env file

File: server/controllers/items.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          id: req.params.id
+        }
+      }).then(function(update) {
+        if (!update) {
+          res.status(500).send({
+            error: 'Update failed'
+          });
+        } else {
+          res.json({
+            isUpdate: true,
```
Comment (andela-gukpere) [inline server/controllers/items.js]: **Code Quality - Nitpick**

The `isUpdate` property is redundant.

File: server/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  var User = require('../schemas/users'),
+    passport = require('passport');
+
+  module.exports = {
+    // login middleware
+    login: function(req, res, next) {
+      passport.authenticate('login', function(err, user) {
+        if (err) {
+          // will generate a 409 error
+          return res.status(409).send({
```
Comment (andela-gukpere) [inline server/controllers/users.js]: **Nitpick**

Status code `409` represents a `CONFLICT` is that our intentation?
From the preceding code block, a login error from passport returns `409` as the first and only error.

File: server/routes/index.js [hunk 4]
```diff
@@ -1,18 +1,35 @@
 module.exports = function(app, config, passport) {
-  require('./users')(app, config, passport);
+  // Authenticate the API routes
+  app.all('/api/*', function(req, res, next) {
```
Comment (andela-gukpere) [inline server/routes/index.js]: **Code Quality - Critical**

For authentication verification, you should add a middleware to the routes that require authentication.
Technically, the middleware should be a function created in the `User` controller, as passed as a middleware were required.

File: server/controllers/users.js (hunk 3 above)
Comment (kn9ts) [inline server/controllers/users.js]: No. It is a internal server error.

**Pull request merged**